pylint/checkers/strings.py

```diff
@@ -305,6 +332,19 @@ def visit_binop(self, node):
                     if key not in required_keys:
                         self.add_message('unused-format-string-key',
                                          node=node, args=key)
+                for key, arg in args.items:
+                    if not isinstance(key, astroid.Const):
+                        continue
+                    format_type = required_types.get(key.value, None)
+                    arg_type = utils.safe_infer(arg)
+                    if (format_type is not None
```

Comment:
Make sure to validate that `arg_type` is valid. `safe_infer` can return either the inferred object or `None` or a singleton `Uninferable`, but you should be able to just do `if arg_type`, since `Uninferable` acts falsey in a boolean context.

Reply:
The check was done in the new function but I agree that doing it inplace is clearer.